src/storage/src/source/mod.rs

```diff
@@ -367,6 +365,17 @@ impl From<anyhow::Error> for SourceReaderError {
     }
 }
 
+pub(crate) type SourceReaderStream<S> = Pin<
```

Comment:
You can simplify this type by a lot, and maybe re-inline using https://docs.rs/futures/latest/futures/stream/type.LocalBoxStream.html